vortex-array/src/arrays/chunked/compute/take.rs

```diff
@@ -24,26 +31,46 @@ impl TakeKernel for ChunkedVTable {
         let indices_mask = indices.validity_mask();
         let indices = indices.as_slice::<u64>();
 
-        let mut chunks = Vec::new();
+        let mut chunks: Vec<ArrayRef> = Vec::new();
         let mut indices_in_chunk = BufferMut::<u64>::empty();
         let mut start = 0;
         let mut stop = 0;
+        let mut builder: Option<Box<dyn ArrayBuilder>> = None;
+        let max_chunks_threshold = array.nchunks() * CHUNK_GROWTH_MULTIPLIER;
+
         // We assume indices are non-empty as it's handled in the top-level `take` function
         let mut prev_chunk_idx = array.find_chunk_idx(indices[0].try_into()?).0;
         for idx in indices {
             let idx = usize::try_from(*idx)?;
             let (chunk_idx, idx_in_chunk) = array.find_chunk_idx(idx);
 
             if chunk_idx != prev_chunk_idx {
-                // Start a new chunk
+                // Check if we're exceeding the threshold
+                if chunks.len() >= max_chunks_threshold && builder.is_none() {
+                    // Initialize builder and canonicalize all existing chunks
+                    let mut new_builder = builder_with_capacity(
+                        &array.dtype().clone().union_nullability(nullability),
+                        indices.len(),
+                    );
+                    for chunk in chunks.drain(..) {
+                        new_builder.extend_from_array(chunk.as_ref());
+                    }
+                    builder = Some(new_builder);
+                }
+
+                // Process the completed chunk
                 let indices_in_chunk_array = PrimitiveArray::new(
                     indices_in_chunk.clone().freeze(),
                     Validity::from_mask(indices_mask.slice(start..stop), nullability),
                 );
-                chunks.push(take(
-                    array.chunk(prev_chunk_idx),
-                    indices_in_chunk_array.as_ref(),
-                )?);
+                let taken = take(array.chunk(prev_chunk_idx), indices_in_chunk_array.as_ref())?;
+
+                if let Some(ref mut b) = builder {
+                    b.extend_from_array(taken.as_ref());
```

Comment:
shame we don't have a take_into method.

Reply:
We deleted it at some point, we should add it back